File: testprojects/csharp/benchmark/domain/locationbased/foursquare/WebSettings.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Igloo.SharpSquare.Core;

namespace CSharpBenchmark.domain.locationbased.foursquare
{
    //Functional requirement 3.2
    //Test case 146: Class domain.locationbased.foursquare,WebSettings is not allowed to use library SharpSquare.dll
    //Result: FALSE
    public class WebSettings
    {
        public void handleCallback()
        {
            try
            {
                string clientId = "CLIENT_ID";
                string clientSecret = "CLIEND_SECRET";
                string redirectUri = "REDIRECT_URI";
                SharpSquare sharpSquare = new SharpSquare(clientId, clientSecret);
                sharpSquare.GetAuthenticateUrl(redirectUri);

            }
            catch (Exception  e)
            {
                Console.WriteLine(e.StackTrace);
            }
        }
    }
}

File: testresources/java/benchmark_2012/domain/locationbased/foursquare/WebSettings.java
package domain.locationbased.foursquare;

import fi.foyt.foursquare.api.FoursquareApi;
import fi.foyt.foursquare.api.FoursquareApiException;
//Functional requirement 3.2
//Test case 146: Class domain.locationbased.foursquare,WebSettings is not allowed to use library foursquareapi.jar
//Result: FALSE
public class WebSettings {
	public void handleCallback() {
		try {
			String code = "1234567890";
			new FoursquareApi("Client ID", "Client Secret", "Callback URL").authenticateCode(code);

		} catch (FoursquareApiException e) {

		}
	}
}